libs/mve/bundle_io.cc

```diff
@@ -480,4 +483,537 @@ save_photosynther_bundle (Bundle::ConstPtr bundle, std::string const& filename)
     out.close();
 }
 
+/* -------------- Support for Colmap --------------- */
+
+// See colmap/src/util/types.h
+typedef uint32_t camera_t;
+typedef uint32_t image_t;
+typedef uint64_t image_pair_t;
+typedef uint32_t point2D_t;
+typedef uint64_t point3D_t;
+
+std::map<camera_t,std::string> camera_model_code_to_name;
+
+void
+define_camera_models() {
+    camera_model_code_to_name.emplace(0, "SIMPLE_PINHOLE");
+    camera_model_code_to_name.emplace(1, "PINHOLE");
+    camera_model_code_to_name.emplace(2, "SIMPLE_RADIAL");
+    camera_model_code_to_name.emplace(3, "RADIAL");
+    camera_model_code_to_name.emplace(4, "OPENCV");
+    camera_model_code_to_name.emplace(5, "OPENCV_FISHEYE");
+    camera_model_code_to_name.emplace(6, "FULL_OPENCV");
+    camera_model_code_to_name.emplace(7, "FOV");
+    camera_model_code_to_name.emplace(8, "SIMPLE_RADIAL_FISHEYE");
+    camera_model_code_to_name.emplace(9, "RADIAL_FISHEYE");
+    camera_model_code_to_name.emplace(10, "THIN_PRISM_FISHEYE");
+}
+
+void
+check_stream(std::ifstream & in, std::string const& filename)
+{
+    if (!in.good())
+        throw util::FileException(filename, std::strerror(errno));
+}
+
+void
+consume_comment_lines(std::ifstream & in)
+{
+    while (in.peek() == '#')
+    {
+        std::string temp;
+        std::getline(in, temp);
+    }
+}
+
+void
+create_camera_info_from_params(CameraInfo& camera_info,
+    std::string const& model,
+    std::vector<float> const& params,
+    uint32_t width,
+    uint32_t height)
+{
+    // https://github.com/colmap/colmap/blob/dev/src/base/camera_models.h
+    // https://github.com/simonfuhrmann/mve/blob/master/libs/mve/camera.cc
+    if (model == "SIMPLE_PINHOLE")
+    {
+        // Simple pinhole: f, cx, cy
+        camera_info.flen = params[0];
+        camera_info.ppoint[0] = params[1] / width;
+        camera_info.ppoint[1] = params[2] / height;
+    }
+    else if (model == "PINHOLE")
```

Comment:
However, I'm not 100% sure that the intrinsic parameter conversion is correct. In particular, I'm wondering if I correctly understood  the meaning of `paspect` ratio.

Reply:
`paspect` is the aspect ratio of the pixels, which should be 1.0 for almost all cameras. It will be >1.0 if pixels are wider than tall.

Reply:
If the focal length is different and x and y direction, that could only mean that pixels have different aspect ratio? If that's the case, your code looks reasonable.

Reply:
According to p. 156 / 157 in "Multiple View Geometry" (Hartley and Zisserman) 
	`fx = f mx` and 
	`fy = f my`, 
where `mx` and `my` are "the number of pixels per unit distance in image coordinates [...] in x and y directions". 
Let us assume that the pixels are wider than tall (i.e. `paspect > 1`).
If pixels are wider than tall, then `my > mx` and `fy > fx`. 
But then `camera_info.paspect = fx / fy` would be `< 1`. 
Thus, I think we have to replace this computation with `camera_info.paspect = fy / fx`, which is `> 1` 
Does that make sense?

Reply:
What you are writing makes sense, the math is sound. But I am getting a little confused now...
Imagine a camera with square pixels, and `fx = fy`. If we now start making pixels less tall, i.e., shrink them vertically, `fy` will get smaller, but `paspect` should get larger, right? Hence `fy / fx` will produce the wrong result? 

As I said, I'm a little confused at this point, I might be missing something.

Reply:
I understand it this way: 
Assuming we have squared pixel (fx = fy) and the size of these squared pixels define the units of some coordinate system (i.e. `my = 1`). In the following `fy'` and `my'` denote the values of the shrinked pixels. If we shrink the pixels vertically, then more pixels actually fit in a unit distance (`my' > 1`). Remember: `mx` and `my` are "the NUMBER of pixels per unit distance in image coordinates [...] in x and y directions". Since `my' > 1`, it follows that `fy' > fy` (i.e. `fy` becomes bigger).
Or in other words, we need more of the shrinked pixels to express the original focal length `f`` (`f = fy / my`).

But I could be wrong here. Since you are the maintainer, you should decide how we handle this.

Side note: I've wrote this importer to compute textured reconstructions of satellite images (the corresponding SfM reconstruction contains camera models with different focal length values fx and fy). With these reconstructions, I hope I am be able to empirically verify this decision :D